eng/imports/HostAgnostic.props

```diff
@@ -12,7 +12,7 @@
     <!-- Path Property: PkgMicrosoft_DiaSymReader_Pdb2Pdb -->
     <PackageReference Include="Microsoft.DiaSymReader.Pdb2Pdb"      ExcludeAssets="all" GeneratePathProperty="true" />
     <PackageReference Include="Microsoft.DotNet.XliffTasks"         PrivateAssets="all" />
-    <PackageReference Include="Microsoft.Net.Compilers.Toolset" />
+    <PackageReference Include="Microsoft.Net.Compilers.Toolset"     PrivateAssets="all" />
```

Comment:
We should re-evaluate whether or not we need this package to begin with. It exists to give you a way to pick up a different version of the compiler than you get with the SDK, and this can be useful if you are stuck with an SDK with a compiler bug or want to use a compiler feature not available through the SDK. However, we don't use lock ourselves to a particular SDK with global.json and frequently have pre-release SDK builds on our systems. As such, the compiler we've chosen here will often be older than the SDK we're using, and sometimes newer, but we're not controlling the relationship between them in any meaningful way.

Reply:
I like that this package lets us dogfoods new compiler bits in a way that works for all developers who might open this repo. If we leave it to the SDK, then developers on older SDKs won't be able to compile.